packages/client/src/runtime/core/engines/binary/BinaryEngine.ts

```diff
@@ -124,6 +122,13 @@ export class BinaryEngine extends Engine<undefined> {
     this.activeProvider = config.activeProvider
     this.connection = new Connection()
 
+    // compute the datasource override for binary engine
+    const dsOverrideName = Object.keys(config.overrideDatasources)[0]
+    const dsOverrideUrl = config.overrideDatasources[dsOverrideName]?.url
+    if (dsOverrideName !== undefined && dsOverrideUrl !== undefined) {
+      this.datasourceOverrides = [{ name: dsOverrideName, url: dsOverrideUrl }]
+    }
```

Comment:
As mentioned before, types were improved to simplify the amount of definitions we maintain for the datasources. Initially the duplicate was more oriented to work directly work with `BinaryEngine`, but today we have three engine implementations, so it made sense to normalize the types. As a side-effect, `BinaryEngine` needs to do a bit of extra work to keep its previous data structures.